src/harness.rs

```diff
@@ -20,187 +17,217 @@ pub enum Error {
                         })
                         .collect::<Result<Vec<Flag>, Error>>()?,
-                )
-            }
-            result_vec[0].clone()
-        }
-        _ => {
-            if test.starts_with("test") {
-                vec![Flag::Inconclusive]
-            } else {
-                return Err(Error::InvalidTestName(test.to_string()));
+                ))
             }
+            result_vec
         }
-    };
-
-    let results = DateRule::new(
-        // TODO: make sure this start time is actually correct
-        Utc.timestamp_opt(cache.start_time.0, 0).unwrap(),
-        cache.period,
-    )
-    .skip(cache.num_leading_points.into())
-    .zip(flags.into_iter())
-    .map(|(time, flag)| SeriesTestResult {
-        time: Some(prost_types::Timestamp {
-            seconds: time.timestamp(),
-            nanos: 0,
-        }),
-        flag: flag.into(),
-    })
-    .collect();
-
-    Ok(ValidateSeriesResponse {
-        test: test.to_string(),
-        results,
-    })
-}
-
-#[allow(clippy::match_single_binding)]
-pub async fn run_test_spatial(
-    test: &str,
-    cache: &DataCache,
-) -> Result<ValidateSpatialResponse, Error> {
-    let flags: Vec<Flag> = match test {
         "buddy_check" => {
             let n = cache.data.len();
 
-            let series_len = cache.data[0].len();
-            let mut result_vec = Vec::with_capacity(series_len);
-            for i in 0..series_len {
+            let series_len = cache.data[0].1.len();
+
+            let mut result_vec: Vec<(String, Vec<Flag>)> = cache
+                .data
+                .iter()
+                .map(|ts| (ts.0.clone(), Vec::with_capacity(series_len)))
+                .collect();
+
+            for i in (cache.num_leading_points as usize)
+                ..(series_len - cache.num_trailing_points as usize)
+            {
                 // TODO: change `buddy_check` to accept Option<f32>?
-                let inner: Vec<f32> = cache.data.iter().map(|v| v[i].unwrap()).collect();
-                result_vec.push(
-                    olympian::buddy_check(
-                        &cache.rtree,
-                        &inner,
-                        &vec![5000.; n],
-                        &vec![2; n],
-                        2.,
-                        200.,
-                        0.,
-                        1.,
-                        2,
-                        &vec![true; n],
-                    )?
+                let inner: Vec<f32> = cache.data.iter().map(|v| v.1[i].unwrap()).collect();
+
+                let spatial_result = olympian::buddy_check(
+                    &cache.rtree,
+                    &inner,
+                    &vec![5000.; n],
+                    &vec![2; n],
+                    2.,
+                    200.,
+                    0.,
+                    1.,
+                    2,
+                    &vec![true; n],
+                )?;
```

Comment:
I've thought about that change, I originally copied the API directly from titanlib, but clippy already complained about this. I wanted to wait until I've implemented more of the spatial test in case there are params in common that would affect the ideal structure (i.e so far we combined data, lats, lons, elevs into one argument, that's shared between both buddy_check and sct).

Reply:
That's alright, my main issue is that I need to open the function signature to understand what all these numbers mean :rofl: but definitely low priority!